examples/machine/im_direct_mpc_curr_ctr.py

```diff
@@ -1,35 +1,39 @@
-#pylint: disable=wrong-import-position
+"""
+Example of controlling an induction machine with direct model predictive control MPC.
+The controller aims to track the stator current reference. The current reference is calculated based
+on a constant stator flux magnitude reference and a torque reference sequence. The machine operates 
+at a constant (nominal) speed.
+"""
 
+#pylint: disable=wrong-import-position
 import sys as system
 import os
-import time
 
 import numpy as np
 
-## ---------------------------------- ##
-# These allow using soft4pes from this folder ##
-# Get the directory of the current file
+## -------------------------------------------------------------------- ##
+# These allow using soft4pes from this folder
+# Get the directory of the current file and add the grandparent directory
+# (soft4pes) to the path
 current_dir = os.path.dirname(os.path.abspath(__file__))
-
-# Get the parent directory
-parent_dir = os.path.dirname(current_dir)
-
-# Add the parent directory to the Python paths
-system.path.append(parent_dir)
-## ---------------------------------- ##
+system.path.append(os.path.abspath(os.path.join(current_dir, '..', '..')))
+## -------------------------------------------------------------------- ##
 
 from soft4pes import model
 from soft4pes.control import mpc
 from soft4pes.utils import Sequence
 from soft4pes.sim import Simulation
 
+# Define base values
 base = model.machine.BaseMachine(Vr=400, Ir=4.4, fr=50, npp=1, pf=0.85)
 
-# The torque reference has two columns, as the Sequence class does not yet support n-dimensional
-# sequences. The reference is saved at the first column.
-T_ref_seq = Sequence(np.array([0, 0.1, 0.1, 0.2]),
-                     np.array([[1, 0], [1, 0], [0.5, 0], [0.5, 0]]))
+# Define torque reference sequence
+T_ref_seq = Sequence(
+    np.array([0, 0.1, 0.1, 0.2]),
+    np.array([1, 1, 0.5, 0.5]),
```

Comment:
I think it would make more sense to have the torque changing from 1 to 0

Reply:
Modified in the next commit